ipwb/indexer.py

```diff
@@ -170,6 +170,83 @@ def indexFileAt(warcPaths, encryptionKey=None,
         print('\n'.join(cdxjLines))
 
 
+def generateCDXJMetadata(cdxjLines=None):
+    metadata = ['!context ["http://tools.ietf.org/html/rfc7089"]']
+    metaVals = {
+        'generator': "InterPlanetary Wayback v.{0}".format(ipwbVersion),
+        'created_at': '{0}'.format(datetime.datetime.now().isoformat())
+    }
+    metaVals = '!meta {0}'.format(json.dumps(metaVals))
+    metadata.append(metaVals)
+
+    return metadata
+
+
+def joinCDXJFiles(cdxjPath1, cdxjPath2, outputFilePath):
+    # CDXJ2 takes precedence in surt uri and datetimes identity
+
+    # Join two files quickly
+    with open(outputFilePath, 'wb') as wfd:
+        for f in [cdxjPath1, cdxjPath2]:
+            with open(f, 'rb') as fd:
+                shutil.copyfileobj(fd, wfd, 1024 * 1024 * 10)
+
+    cdxjLines = ''
+    with open(outputFilePath, 'r') as wfd:
+        cdxjLines = wfd.read().split('\n')
+
+        # De-dupe and sort, needed for CDXJ adherence (pulled from indexer.py)
+        cdxjLines = list(set(cdxjLines))
+        cdxjLines.sort()
+
+        cdxjLines[:] = [line for line in cdxjLines
+                        if len(line) > 0 and line[0] != '!']
+
+        # Prepend metadata
+        cdxjMetadataLines = generateCDXJMetadata(cdxjLines)
+        cdxjLines = cdxjMetadataLines + cdxjLines
+
+        cdxjLines = '\n'.join(cdxjLines)
+
+    with open(outputFilePath, 'w') as wfd:
+        wfd.write(cdxjLines)
+
+
+def isCDXJMetadataRecord(cdxjLine):
```

Comment:
This function can be as simple as:

```py
    return bool(cdxjLine.strip()) and cdxjLine.strip().startswith('!')
```

Reply:
That refactoring can been performed after the PR. This function existed prior to this PR and is not original code. Please focus on the changes in this PR.

Reply:
In fact there can be an event shorter syntax:

```py
    return cdxjLine.strip()[:1] != '!'
```

Reply:
I agree but again, this function is not introduced in this PR.

Reply:
With the changes suggested in the replay and the removal of a big function in the indexer, the code is going to be a lot lighter, simpler to follow, easier to maintain, and functionally the same.